\textbf{Note: }\textit{The majority of the work presented in this chapter was published in two independent pieces of work. All work relating to the globular targets was published by \textcite{Simkovic2016-wk}, and a great majority of work relating to the transmembrane targets by \textcite{Thomas2017-sh}. As such, this chapter consists of extracts from both publications with additional information where appropriate. Text duplicated from either publication was written by Felix Simkovic, all other elements were adapted.}

\section{Introduction}
The introduction of residue-residue contacts as distance restraints in \textit{ab initio} protein structure prediction has proven to be a highly successful approach to limiting the conformation search space thereby enabling successful fold predictions of larger and more \textbeta-rich protein structures \cite[e.g.,][]{Marks2011-os,Michel2014-eg,Kosciolek2014-bt,Ovchinnikov2015-tn,Ovchinnikov2016-jj,Michel2017-xh,De_Oliveira2017-sg,Ovchinnikov2017-nd,Wang2017-rx}. In AMPLE, such proteins have historically proven the most difficult targets \cite{Bibby2012-lm}. Furthermore, the initial AMPLE study by \textcite{Bibby2012-lm} focused solely on globular targets, whilst \textcite{Thomas2017-qu} focused only much later on transmembrane protein targets. Contact information was shown to be useful for both target classes, and thus should prove invaluable to AMPLE users.

Since the application of much more accurate \textit{ab initio} protein structure prediction --- obtained by restraining the conformational search space with residue-residue contacts --- has not yet been explored, this initial study examines the impact on AMPLE performance of contact predictions. The aim is to extend the target tractability with particular focus on larger and more \textbeta-rich protein structures.

\section{Materials \& Methods}
\subsection{Target selection}
In this study, targets from the ORIGINAL and TRANSMEMBRANE datasets were used. This resulted in a final set of 21 globular and 17 transmembrane protein targets. For details on how the targets were selected refer to \cref{sec:methods_dataset_original,sec:methods_dataset_transmembrane}, and for details on each target refer to \cref{table:appendix_dataset_transmembrane,table:appendix_dataset_original}.

\subsection{Contact prediction} \label{sec:ample_proof_conpred}
For all globular targets, one contact map was predicted with the fully automated metapredictor PCONSC2 v1.0 \cite{Skwark2014-qp}. In summary, four \gls{msa}s were generated with each of JACKHMMER v3.1b2 \cite{Johnson2010-uz} against the \texttt{uniref100} v2015-10 database and HHBLITS v2.0.15 \cite{Remmert2011-kt} against the \texttt{uniprot20} v2013-03 database \cite{Bateman2017-pb} at E-value cutoffs of 10e\textsuperscript{-40}, 10e\textsuperscript{-10}, 10e\textsuperscript{-4} and 1. Each \gls{msa} was analysed with PSICOV v2.13b3 \cite{Jones2012-ks} and PLMDCA v2 \cite{Ekeberg2014-kf} to produce 16 individual contact predictions. All 16 predictions and per-target PSIPRED v3 \cite{Jones1999-ed} secondary structure prediction, NETSURFP v1.0 \cite{Petersen2009-wy} solvent accessibility information and HHBLITS v2.0.15 \cite{Remmert2011-kt} sequence profile were provided to the PCONSC2 deep learning algorithm \cite{Skwark2014-qp} to identify protein-like contact patterns. The latter produced a final predicted contact map for each target sequence.

An additional contact map for \textbeta-structure containing targets was predicted using CCMPRED v0.3 \cite{Seemayer2014-zp} and reduced to \textbeta-sheet contact pairs using the CCMPRED-specific filtering protocol BBCONTACTS v1.0 \cite{Andreani2015-qn}. Each \gls{msa} for CCMPRED contact prediction was obtained using HHBLITS v2.0.15 \cite{Remmert2011-kt}. This entailed two sequence search iterations with an E-value cutoff of 10\textsuperscript{-3} against the \texttt{uniprot20} v2013-03 database \cite{Bateman2017-pb} and filtering to 90\% sequence identity using HHFILTER v2.0.15 \cite{Remmert2011-kt} to reduce sequence redundancy in the \gls{msa}. Besides the contact matrix as input, BBCONTACTS requires a secondary structure prediction and an estimate of the \gls{msa} diversity. The secondary structure prediction was taken from the PCONSC2 step whilst the diversity factor was calculated using \cref{eq:methods_alndiversity}.

For each transmembrane protein target, a \gls{msa} was generated using HHBLITS v2.0.16 \cite{Remmert2011-kt} against \texttt{uniprot20} v2016-02 database \cite{Bateman2017-pb}. Three search iterations were selected at an E-value cutoff of 1e\textsuperscript{-3} and minimum coverage with the target sequence of 60\%. Contact predictions for each transmembrane target were obtained using the metapredictor METAPSICOV v1.04 \cite{Jones2015-vq}, which in turn used the contact prediction algorithms CCMPRED v0.3.2 \cite{Seemayer2014-zp}, FREECONTACT v1.0.21 \cite{Kajan2014-bx} and PSICOV v2.1b3 \cite{Jones2012-ks}. Additionally, a set of contacts was also predicted using the MEMBRAIN server v2015-03-15 \cite{Yang2013-bf}.

\subsection{Contact-to-restraint conversion} \label{sec:bbcontacts_addition}
For all targets, the predicted contact maps were converted to ROSETTA restraints to guide \textit{ab initio} structure prediction. The FADE energy function was used to introduce a restraint in ROSETTA's folding protocol. The implementation described by \textcite{Michel2014-eg} was used, which defined a contact to be formed during folding if the participating C\textbeta\ atoms (C\textalpha\ in case of glycine) were within 9\AA\ of one another. The top-$L$ ($L$ corresponds to the number of residues in the target sequence) contact pairs were converted to ROSETTA restraints, and if satisfied a ``squared-well'' bonus of -15.00 added to the energy function.

Additionally, all \textbeta-containing targets were subjected to a further conversion step in a separate condition. The approach of adding BBCONTACTS restraints to a previous prediction is outlined in \cref{sec:methods_bbcontacts_addition}.

\subsection{\textit{Ab initio} structure prediction}
Fragments for all targets were selected using the \texttt{make\_fragments.pl} script shipped with ROSETTA. To ensure no closely homologous fragments were included in the fragment libraries, the \texttt{-nohoms} flag was set. This performs a PSIBLAST search to identify sequence homologs, whose corresponding \gls{pdb} IDs are subsequently excluded from the fragment search. Each target's secondary structure prediction was provided to the fragment picker using the \texttt{-psipredfile} argument. The fragment libraries, contact restraints and secondary structure prediction were provided to the ROSETTA \texttt{AbinitioRelax} protocol \cite{Rohl2004-dj} to predict 1,000 decoys per target. ROSETTA options were chosen according to the default protocol in AMPLE v1.0 \cite{Bibby2012-lm}. ROSETTA v2015.05.57576 was used for globular targets and v2015.22.57859 for transmembrane ones for all ROSETTA-related protocols.
 
\subsection{Molecular Replacement in AMPLE}
All generated decoys were subjected to AMPLE v1.0 \cite{Bibby2012-lm} for ensemble search model generation. 

All transmembrane protein targets were processing using AMPLE's default parameters. \Gls{mr} trials were performed with software versions shipped in CCP4 v6.5.13 \cite{Winn2011-xe}, with the exception of SHELXE v2014/14 \cite{Thorn2013-le} and ARP/wARP v7.5 \cite{Cohen2007-wg}.

All globular protein targets were subjected to AMPLE with two deviations from the default parameters. The \texttt{-use\_scwrl} was set to subject all decoys to side-chain remodelling using SCWRL4 \cite{Krivov2009-ex}. Furthermore, the number of clusters to trial was set increased from one to three via the \texttt{-num\_clusters}parameter. All \gls{mr} trials were performed with the version of software shipped with CCP4 v6.5.15 \cite{Winn2011-xe}.

All \gls{mr} solutions were assessed for success using the criteria described in \cref{sec:methods_mr_success}.

\section{Results}
In this study, the application of residue-residue contact predictions to \textit{ab initio} protein structure prediction and subsequently \gls{mr} was investigated. This proof-of-concept work is based on two datasets covering a range of globular and transmembrane protein targets. At the time of conducting this study, state-of-the-art contact prediction algorithms were applied to obtain the best possible contact predictions to see how much AMPLE performance could be improved \cite{Bibby2012-lm}.

\subsection{Residue-residue contact prediction}
Accurate coevolution-based residue-residue contact prediction is highly dependent on the availability of many divergent homologous sequences. As such, it is important to validate that the selected targets in this study satisfy such requirement.

The depth of \gls{msa}s obtained for each target sequence suggests that sufficient numbers of divergent homologous sequences are available. Across all globular targets, the minimum alignment depth is obtained for Galectin-3 domain (\gls{pdb} ID: 1kjl) with 679 effective sequences and the maximum for G-protein Arf6-GDP (\gls{pdb} ID: 1e0s) with 1,897 effective sequences (\cref{fig:ample_proof_globpreccomp}). The median alignment depth for all globular targets is over 1,000, which is beyond the often suggested threshold of 200 sequences \cite{Simkovic2017-xs}. The \gls{msa}s for all transmembrane protein targets also surpass this threshold comfortably. The median alignment depth is much higher than for globular targets with 1,878 sequences (\cref{fig:ample_proof_tmpreccomp}). The minimum, which was obtained for Sensory rhodopsin II (\gls{pdb} ID: 1gu8), is 692 sequences and the maximum for the sequence of Rhomboid protease GLPG (\gls{pdb} ID: 2xov) is 6,583.

\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_globpreccomp.pdf}
        \caption{}
        \label{fig:ample_proof_globpreccomp}
    \end{subfigure}
    
    \begin{subfigure}[b]{\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_tmpreccomp.pdf}
        \caption{}
        \label{fig:ample_proof_tmpreccomp}
    \end{subfigure}
    
    \caption[Alignment depth and contact precision analysis of globular and transmembrane protein targets]{Alignment depth and contact precision analysis of (a) globular and (b) transmembrane protein targets. Contact predictions were obtained with several contact prediction algorithms. Precision scores were calculated for the top-$L$ contact pairs. JACKHMMER and HHBLITS alignments for PSICOV and PLMDCA contact predictions in (a) were obtained with E-value $1e^{-4}$.} 
    \label{fig:ample_proof_preccomp}
\end{figure}

In coevolution-based contact prediction, the precision depends on alignment depth. Despite sufficient number of effective sequences across all targets, findings in this study suggest that some (meta-)predictors cannot fully utilise greater alignment depths to correct contact pairs (\cref{fig:ample_proof_preccomp}). 

PCONSC2 --- a metapredictor using eight starting alignments and two contact predictors --- outperforms its individual parts for almost all globular targets (\cref{fig:ample_proof_globpreccomp}). Although only four individual components are shown in \cref{fig:ample_proof_globpreccomp}, the pattern translates across all 16 individual predictions per target. Such results suggest that precision greatly depends on the tool used to identify and select homologous sequences for the \gls{msa}. A closer inspection of mean precision scores resulting from HHBLITS- and JACKHMMER-based alignments shows higher precision scores for top-$L$ contact pairs based on the former alignments (\cref{table:ample_proof_pconsc2}). Nevertheless, the Machine Learning approach in PCONSC2 to combine more and less precise individual predictions results in superior precision in the output (\cref{table:ample_proof_pconsc2}). No correlation could be established between alignment depth and precision for either individual predictors or the metapredictor PCONSC2 (\cref{fig:ample_proof_globpreccomp}). 

\begin{table}[H]
    \centering
    \caption[Summary of raw conact prediction precision values in PCONSC2]{Summary of mean PCONSC2 raw contact prediction precision based on JACKHMMER and HHBLITS alignments and PSICOV, PLMDCA and PCONSC2 coevolution-based contact prediction.}
    \label{table:ample_proof_pconsc2}
    \begin{tabularx}{\textwidth}{b b s s s s}
        \hline
        \multicolumn{2}{c}{\textbf{Contact prediction}} & \multicolumn{4}{c}{\textbf{Alignment E-value cutoff}} \\
        \cline{3-6} & & $1e^{0}$ & $1e^{-4}$ & $1e^{-10}$ & $1e^{-40}$ \\ 

        \hline
        \multirow{2}{1em}{PSICOV} & JACKHMMER  & 0.240 & 0.239 & 0.213 & 0.167 \\
                                  & HHBLITS    & 0.439 & 0.435 & 0.354 & 0.209 \\
        \hline
        \multirow{2}{1em}{PLMDCA} & JACKHMMER  & 0.293 & 0.288 & 0.252 & 0.140 \\ 
                                  & HHBLITS    & 0.545 & 0.530 & 0.447 & 0.224 \\
        \hline
        \hline
        PCONSC2                   &            & \multicolumn{4}{c}{0.667}     \\
        \hline
    \end{tabularx}
\end{table}

Contacts for transmembrane protein targets in this study were predicted with the metapredictor METAPSICOV and the transmembrane-specific predictor MEMBRAIN.

METAPSICOV STAGE 1 and STAGE 2 predictions outperform MEMBRAIN in nine and ten cases, respectively, whilst MEMBRAIN outperforms METAPSICOV for the rest (\cref{fig:ample_proof_tmpreccomp}). The METAPSICOV algorithm utilises the raw predictions by CCMPRED, FREECONTACT and PSICOV to generate its STAGE 1 and STAGE 2 predictions. METAPSICOV STAGE 1 predictions are near identical to CCMPRED, whereby 12 of 13 targets show an absolute $\Delta_{precision}$ of less than 0.05 (\cref{fig:ample_proof_tmpreccomp}). This similarity does not propagate to METAPSICOV STAGE 2 predictions with only a single target showing such similar precision values (\cref{fig:ample_proof_tmpreccomp}). Amongst the three raw predictors used by METAPSICOV, FREECONTACT performs by far the worst with a mean precision of 0.09 across all transmembrane targets. PSICOV shows similar trend to CCMPRED when assessed by target, which results in a mean absolute $\Delta_{precision}$ of 0.10.

The addition of BBCONTACTS contact pairs to improve structure prediction accuracy for \textbeta-structure containing targets is a novel aspect introduced in this study. The initial step of the addition of BBCONTACTS contact pairs included the filtering of one- and two-pair strand contacts from the original BBCONTACTS list, since those contained high numbers of false positives (Jessica Andreani, personal communication). The findings in this study confirm this for all \textbeta-structure containing targets. Precision values improved for all targets with changes ranging from 0.01 to 0.14 whilst retaining on average 80\% of all contacts. Filtered BBCONTACTS predicted contact maps were combined with other predicted contact maps, i.e. PCONSC2, to either upweight or add contact pairs. Findings in this study highlight that upweighted contact pairs are more precise than ones to be added. The minimum precision score for a set of upweighted contacts is 0.72 for 29 contact pairs and the maximum of 1.00 for up to 27 contact pairs. In comparison, novel BBCONTACTS predicted contact pairs not present in the base predicted contact map range in precision scores from 0.22 (nine contacts) to 0.76 (21 contacts). 

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_bbaddon.pdf}
    \caption[Evaluation of BBCONTACTS contact pairs]{Evaluation of BBCONTACTS contact pairs. (a) Precision evaluation of PCONSC2 predicted contact map with and without BBCONTACTS. (b) Precision evaluation of BBCONTACTS contact pairs split by status of presence or absence in the base PCONSC2 contact list. Numbers besides each marker indicate the number of contacts. The order scatter points is identical between both subplots and based on the PCONSC2 precision values (x-axis) in (a).}
    \label{fig:ample_proof_bbaddon}
\end{figure}

Despite the high precision of BBCONTACTS contact pairs, the merge of such pairs with top-$L$ PCONSC2 contact pairs results in an expected loss in precision for the resulting contact set (\cref{fig:ample_proof_bbaddon}). True positive contacts, which dominate the BBCONTACTS contact set, are usually also predicted by PCONSC2, and thus upweighted (\cref{fig:ample_proof_bbaddon}). Since upweighting does not affect the precision, the value remains unaffected after this procedure. However, contact pairs unique to BBCONTACTS contain more false positives. Once added to the base PCONSC2 contact list, these contacts therefore reduce the precision value (\cref{fig:ample_proof_bbaddon}). Either subset of BBCONTACTS contacts does not show any correlation between the number it contains and its precision. The fold of the target does not show any clear distinction between better and worse sets of contacts either (\cref{fig:ample_proof_bbaddon}).

\subsection{Protein structure prediction}
Predicted contact information is particularly useful to limit the conformation search space in \textit{ab initio} protein structure prediction. Since such predictions are the basis for AMPLE studies presented in this thesis, it is important to analyse the improvement in decoy quality.

Globular protein targets benefit greatly from the addition of PCONSC2 residue contacts. All but one target see median \gls{tmscore} improvements of at least 0.05 when comparing contact-assisted PCONSC2 decoys with simple ROSETTA decoys (\cref{fig:ample_proof_comproscon}). The greatest improvement over 1,000 decoys was achieved for Oxy-myoglobin (\gls{pdb} ID: 1a6m) with an improvement in median \gls{tmscore} of 0.42.  The decoys for Ankyrin (\gls{pdb} ID: 2qyj) show a minor decrease in median \gls{tmscore} of 0.04; however, the median \gls{tmscore} for ROSETTA decoys is 0.78, and thus a minor decrease may be negligible. 

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_comproscon.pdf}
    \caption[Effect of contact distance restraints on \textit{ab initio} decoy quality]{Effect of contact distance restraints on \textit{ab initio} decoy quality by comparison of unrestrained (\textit{no contacts}) and contact-restrained (\textit{contacts}) median TM-scores for 1,000 decoys per target. Colours indicate the target fold and symbols the contact prediction algorithm.}
    \label{fig:ample_proof_comproscon}
\end{figure}

Previously, \textit{ab initio} protein structure prediction for globular targets was greatly limited by target fold and chain length. The addition of residue-residue contacts enhances decoy quality primarily for \textalpha-helical and mixed \textalpha-\textbeta\ protein targets (\cref{fig:ample_proof_globtmcomp}). Whilst only one all-\textalpha\ target has more than 50\% native-like decoys in its ROSETTA decoy set, five targets surpass this threshold when PCONSC2 contact data is used to restrain the folding procedure. Similarly, the median \gls{tmscore} of no mixed \textalpha-\textbeta\ target decoy set surpasses the \gls{tmscore} threshold of 0.5 with ROSETTA decoys compared to one for PCONSC2 decoys with three further ones greater than 0.4. All-\textbeta\ targets also benefit from the addition of contact restraints, although decoy sets do not surpass the native-like threshold by median \gls{tmscore} (\cref{fig:ample_proof_globtmcomp}). Larger targets do not benefit any more than smaller targets from the addition of residue contacts to the structure prediction protocol. The only real exception to this are the decoys for the CheB methylesterase domain (\gls{pdb} ID: 1chd), for which the majority of ROSETTA decoys are almost random-like whilst PCONSC2 decoys are native-like (\cref{fig:ample_proof_globtmcomp}).

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_globtmcomp.pdf}
    \caption[TM-score comparison for globular targets separated by fold]{\gls{tmscore} comparison for globular targets separated by fold and ordered by target chain length. Median \gls{tmscore}s for 1,000 decoys generated with simple ROSETTA (orange) or contact-assisted ROSETTA (blue) runs. White numbers in each row correspond to the target chain length. Bars surpassing the dark gray line indicate that the majority of structures are better than random, whilst the light gray line indicates that the majority of structures are native-like \cite{Xu2010-kr}.}
    \label{fig:ample_proof_globtmcomp}
\end{figure}

The enhancement of \textbeta-structure specific contact pairs is an important part of this study. Previously, the precision of the added BBCONTACTS contact pairs has been demonstrated. The next essential step is to explore how the BBCONTACTS supplement enhances or degrades decoy quality after ROSETTA \textit{ab initio} protein structure prediction. Given 13 \textbeta-structure containing targets, eight targets achieve better overall decoy quality with added BBCONTACTS (\cref{fig:ample_proof_comprosconbb}). The smallest improvement is observed for target 1e0s with 0.01 \gls{tmscore} units, whilst the largest for target 1eaz with 0.05 units. The remaining five targets --- \gls{pdb} IDs 1chd, 1bdo, 1npu, 4u3h and 1tjx --- saw decreases in median \gls{tmscore} up to 0.03 when BBCONTACTS contact pairs were added as restraints (\cref{fig:ample_proof_comprosconbb}). No clear difference between fold classes, i.e. mixed \textalpha-\textbeta\ or all-\textbeta\ targets, could be observed, although mixed \textalpha-\textbeta\ targets do show slightly greater extremes (\cref{fig:ample_proof_comprosconbb}).

\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_comprosconbb.pdf}
        \caption{}
        \label{fig:ample_proof_comprosconbb}
    \end{subfigure}
    
    \begin{subfigure}[b]{\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_bbsat.pdf}
        \caption{}
        \label{fig:ample_proof_bbsat}
    \end{subfigure}
    
    \caption[Decoy analysis of effects of BBCONTACTS contact addition]{(a) \Gls{tmscore} comparison for \textbeta-structure containing globular targets separated by fold and ordered by the difference in median \gls{tmscore} between PCONSC2 and PCONSC2+BBCONTACTS decoys. Positive values indicate a better median \gls{tmscore} in favour of PCONSC2+BBCONTACTS decoys, whilst negative values those for PCONSC2. \Gls{pdb} IDs are provided alongside each bar. (b) Satisfaction of BBCONTACTS contact pairs in decoys with added \textbeta-structure contact restraints (PCONSC2 + BBCONTACTS) and those without (PCONSC2).}
\end{figure}

An analysis of the satisfaction of BBCONTACTS contact pairs in decoys where extra \textbeta-sheet contact pairs were used as restraints compared those where they were not highlights a greater satisfaction in the former. This indicates that the added and upweighted BBCONTACTS \textbeta-structure contact restraints enhance the formation of \textbeta-sheets in the resulting decoys, which would explain the overall improved decoy quality for more than half the targets outlined previously. A separation of contact satisfaction by upweighted and added BBCONTACTS contact pairs indicates that the upweighting has less effect compared to the addition (\cref{fig:ample_proof_bbsat}). Although the former shows a marginal improvement in BBCONTACTS contact satisfaction of decoys without upweighted restraints, the difference is minimal. In comparison, PCONSC2 decoys without the added BBCONTACTS restraints show less satisfaction for such contacts, indicating that they did not form as often compared to PCONSC2+BBCONTACTS. In combination with the upweighting, this explains that \textbeta-rich regions are predicted more accurately when BBCONTACTS contact pairs supplement PCONSC2 contacts.

Transmembrane protein targets were modelled using residue-residue contact predictions derived with CCMPRED, MEMBRAIN and METAPSICOV STAGE1. A ROSETTA benchmark was also run to compare contact-assisted decoys to the current norm. Findings in this study highlight the much improved decoy quality for almost all targets when contact information is used to reduce the conformational sampling space (\cref{fig:ample_proof_comproscon}). Across all methods, only the decoys for ATP synthase subunit C (\gls{pdb} ID: 2wie) suffered from the addition of contact restraints during \textit{ab initio} protein structure prediction. ROSETTA generated decoys with median \gls{tmscore} of greater than 0.6 when no contact restraints were used. This contrasts strongly with contact-assisted decoy sets, for which only METAPSICOV STAGE1 predictions yielded overall native-like decoys, i.e. median TM-score of greater than 0.5. 

A split for decoy quality comparison between no-contact and contact-assisted decoy sets by contact prediction algorithm shows that CCMPRED contacts are not sufficiently precise to always improve decoy quality. Four out of 13 targets are predicted more accurately without CCMPRED contact information (\cref{fig:ample_proof_tmtmcomp}). In comparison, MEMBRAIN and METAPSICOV STAGE1 contact predictions result in enhanced decoy quality to the extend that only one decoy set is worse than their no-contact counterpart (\cref{fig:ample_proof_tmtmcomp}). Most notably, either of the three contact predictions per target performs better for certain targets. The most extreme example may be the decoy sets for Bacteriorhodopsin (\gls{pdb} ID: 3hap) for which CCMPRED contacts result in decoy quality degradation of 0.06, METAPSICOV STAGE1 in a slight improvement of 0.06 and MEMBRAIN in an improvement of 0.28 \gls{tmscore} units. This translates into absolute decoy counts with native-like fold --- i.e., \gls{tmscore} $\geq 0.5$ --- of the following: 274 for decoys without contact guidance, 289 for CCMPRED contact guidance, 538 for METAPSICOV STAGE1 contact guidance, and 996 for MEMBRAIN contact guidance. Similar examples exist (e.g., \gls{pdb} IDs 1gu8, 3rlb or 4dve in \cref{fig:ample_proof_tmtmcomp}) and highlight that no single method yields the best decoys in all circumstances.

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_tmtmcomp.pdf}
    \caption[TM-score difference between contact-assisted and simple ROSETTA decoys]{TM-score difference between contact-assisted and simple ROSETTA decoys for transmembrane protein targets. Positive $\Delta_{TMscore}$ values indicate more accurate contact-assisted decoys, whilst negative values better decoys without the addition of contacts. $\Delta_{TMscore}$ values were computed by median \gls{tmscore}. Contact restraints were obtained with (a.) CCMPRED, (b.) MEMBRAIN, and (c.) METAPSICOV STAGE1.}
    \label{fig:ample_proof_tmtmcomp}
\end{figure}

\subsection{Molecular Replacement}
The most important aspect of this study is the impact of contact-assisted decoys in AMPLE-\gls{mr}. AMPLE is primarily limited by a target's chain length and fold, which typically cannot exceed 150 residues, and performs poorly for \textbeta-rich folds \cite{Bibby2012-lm}. Findings presented in previous sections of this chapter outlined improvements in overall decoy quality when contact information was used as distance restraints in \textit{ab initio} protein structure prediction. However, it is yet to be seen how the improved decoy quality translates into \gls{mr} structure solutions.

\subsubsection{Globular protein targets}
Structure solutions were attempted for a total of 21 globular targets. Simple ROSETTA decoys --- those without contact restraints and AMPLE's current default --- resulted in nine structure solutions (\cref{fig:ample_proof_mrglob}). The addition of PCONSC2 contact-restraints to the structure prediction procedure improved decoy quality to achieve four additional structure solutions. However, the structure of the N-terminal region of P67Phox (\gls{pdb} ID: 1hh8) was not solved when PCONSC2-restrained decoys were used compared to simple ROSETTA ones. The addition of BBCONTACTS distance restraints to upweight and supplement PCONSC2 contacts enabled a further unique solution for the Phosphoinositol (3,4)-bisphosphate PH domain (\gls{pdb} ID: 1eaz) (\cref{fig:ample_proof_mrglob}).

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_mrglob.pdf}
    \caption[Structure solution summary for globular targets]{Summary of structure solutions obtained with AMPLE using no-contact ROSETTA, PCONSC2-contact-restraint-assisted ROSETTA and PCONSC2 + BBCONTACTS-assisted ROSETTA decoy sets. Empty markers indicate unsolved targets. Filled markers highlight the minimum decoy set by complexity of the prediction procedure to obtain \gls{mr} structure solution. Marker shape distinguishes the fold class.}
    \label{fig:ample_proof_mrglob}
\end{figure}

The majority of structure solutions were obtained for all-\textalpha\ targets in the dataset, with a total of eight structure solutions (\cref{fig:ample_proof_mrglob}). Seven of those eight structure solutions were achieved with unrestrained ROSETTA decoys, with target chain lengths up to 213 residues. The largest target in the globular dataset, and the only all-\textalpha\ target that required residue contacts, totals 221 residues in target chain length, which exceeds AMPLE's previously benchmarked limits for globular targets greatly \cite{Bibby2012-lm}. In comparison to all-\textalpha\ targets, \textbeta-structure containing proteins require the contact restraints to result in sufficiently accurate decoys for \gls{mr}. Across all-\textbeta\ and mixed \textalpha-\textbeta\ targets, only two structure solutions were obtained with unrestrained ROSETTA decoys. This contrasts to an additional three targets when PCONSC2 restraints were used during \textit{ab initio} structure prediction. Furthermore, the addition of BBCONTACTS contact restraints enabled an additional structure solution, yielding much greater success for \textbeta-structure containing protein targets compared to the previous default. Structure solutions for \textbeta-containing targets were obtained for target chain lengths up to 141 residues (\cref{fig:ample_proof_mrglob}).

In this study, two exceptional cases exemplify the application of contact predictions and their benefit to \gls{mr}. The first, the structure solutions for 4-hydroxybenzoyl CoA thioesterase (\gls{pdb} ID: 1lo7) based on AMPLE ensemble search models derived from the PCONSC2 and PCONSC2+BBCONTACTS decoy sets. Without contact restraints, AMPLE search models do not accurately represent the target fold (\cref{fig:ample_proof_1lo7_rosetta}). In comparison, precise residue-residue contacts primarily restraining the large \textbeta-sheet yield decoys of sufficient quality to achieve \gls{mr} structure solution with both PCONSC2 (\cref{fig:ample_proof_1lo7_pconsc2}) and PCONSC2+BBCONTACTS (\cref{fig:ample_proof_1lo7_pconsc2bbcontacts}) decoy sets.

\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{0.32\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_1lo7_rosetta_c1_tl50_r3_polyAla.png}
        \caption{}
        \label{fig:ample_proof_1lo7_rosetta}
    \end{subfigure}
    %
    \begin{subfigure}[b]{0.32\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_1lo7_pconsc2_c2_tl55_r3_reliable.png}
        \caption{}
        \label{fig:ample_proof_1lo7_pconsc2}
    \end{subfigure}
    % 
    \begin{subfigure}[b]{0.32\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_1lo7_pconsc2bbcontacts_c3_tl50_r2_reliable.png}
        \caption{}
        \label{fig:ample_proof_1lo7_pconsc2bbcontacts}
    \end{subfigure}

    \caption[Structural superposition of three search models for target 1lo7]{Structural superposition of the (a) ROSETTA (C\textalpha\ \gls{rmsd} 2.814 \AA; ensemble contains two structures), (b) PCONSC2 (C\textalpha\ \gls{rmsd} 1.748 \AA; 30 members) and (c) PCONSC2+BBCONTACTS (C\textalpha\ \gls{rmsd} 1.760 \AA; 15 members) search-model ensembles for 4-hydroxybenzoyl CoA thioesterase (\gls{pdb} ID: 1lo7). Examples are the highest scoring search models based on SHELXE \gls{cc} score, with only (b) and (c) leading to successful \gls{mr} structure solutions. Search models are shown as tubes and crystal structures as cartoons. (a) and (c) are 50\% of the target sequence, while (b) is 55\%. The colour scale illustrates the pairwise C\textalpha\ \gls{rmsd} between each search-model ensemble (represented by its first member) and the crystal structure, with blue representing the minimum C\textalpha\ \gls{rmsd} and red the maximum. Unaligned residues are coloured grey.}
    \label{fig:ample_proof_example_1lo7}
\end{figure}

The second exceptional case, \gls{pdb} ID 1e0s, does not yield any \gls{mr} structure solution with either decoy set according to the stringent criteria for \gls{mr} success applied in this study (see \cref{sec:methods_mr_success}). However, a \gls{rio} analysis of PHASER solutions, i.e. after \gls{mr}, indicates that some PCONSC2 and PCONSC2+BBCONTACTS AMPLE search models were placed partially correctly (\cref{fig:ample_proof_example_1e0s}). For the top PCONSC2 search model, 40\% (12 residues) of the search model residues are correctly superimposed, albeit out of register on the target structure (PHASER \gls{tfz}=4.7, PHASER \gls{llg}=16) (\cref{fig:ample_proof_1e0s_pconsc2}). For the top PCONSC2+BBCONTACTS search model, 77\% (30 residues) of the search model were superimposed in an in-register fashion (PHASER \gls{tfz}=5.3, PHASER \gls{llg}=17) (\cref{fig:ample_proof_1e0s_pconsc2bbcontacts}). For the latter, expert manual intervention might allow structure determination, but in this case the correct solution was not prominent in the list of \gls{mr} placements. Nevertheless, it is clear that even when overall structure solution was not automatically achieved the PCONSC2+BBCONTACTS search model provided better results which might be recoverable as successes in the future as \gls{mr} and post-\gls{mr} software improves still further.

\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{0.48\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_1e0s_pconsc2.png}
        \caption{}
        \label{fig:ample_proof_1e0s_pconsc2}
    \end{subfigure}
    %
    \begin{subfigure}[b]{0.48\textwidth}
        \centering
        \includegraphics[width=\textwidth]{ample_proof_1e0s_pconsc2bbcontacts.png}
        \caption{}
        \label{fig:ample_proof_1e0s_pconsc2bbcontacts}
    \end{subfigure}

    \caption[Top-PHASER solutions for target 1e0s]{Top-PHASER solutions for \gls{pdb} ID 1e0s based on \gls{rio} scores for (a) PCONSC2 (\gls{rio} score 12) and (b) PCONSC2+BBCONTACTS (\gls{rio} score 30) search models. Search-model colour coding indicates useful superposition of residues by in- (green) or out-of-sequence register (blue) residues as well as misplaced (red) residues. The addition of BBCONTACTS restraints produced a more accurate model with correctly placed \textbeta-strands that was placed correctly. Both structures are shown in cartoon representation with the crystal structure shown as a transparent cartoon. Unaligned reference crystal structure residues are coloured grey.}
    \label{fig:ample_proof_example_1e0s}
\end{figure}

With much improved decoy quality deriving from the use of predicted residue restraints to guide \textit{ab initio} structure prediction, the question arises whether AMPLE's existing cluster-and-truncate approach remains the most suitable for obtaining a conserved, native-like core from the decoys found in the largest clusters. For globular targets solved using simple ROSETTA decoys, certain features throughout AMPLE's cluster-and-truncate approach typically correlated with eventual success in structure solution \cite{Bibby2012-lm}. In general, the greater the number of decoys in the largest cluster the more likely the success was with derived search models. Truncation removed structurally variant parts leading to smaller more accurate ensemble subsets of the cluster decoys. Although successful search models were found at every truncation interval, the majority were derived with search models containing around 30 residues. Lastly, each of the potential nine search models derived at each truncation level (three subclustering radii with three side chain treatments each) can lead to non-redundant structure solutions. Similar observations, particularly with respect to the most successful search model size range were made for other target classes \cite{Thomas2017-sh, Thomas2015-wu} and for \textit{ab initio} decoys made with QUARK \cite{Keegan2015-zb}.

A size comparison of the largest clusters of ROSETTA and PCONSC2+BBCONTACTS (or PCONSC2 for all-\textalpha) decoys indicated a median increase of 122 decoys per cluster in the latter. All cluster sizes increased except for target 2qyj. More accurate \textit{ab initio} decoys are directly linked to larger cluster sizes because of the associated increase in convergence \cite{Simons1997-oe}. Here, as expected, the largest cluster contains better than average quality decoys but the size of the largest cluster does not link to the total number of successful search models.  

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_mrglobtl.pdf}
    \caption[Effect of progressive truncation on RMSD of ensemble centroid]{The percentage of sequence in the search model is mapped against the \gls{rmsd} over all C\textalpha\ atoms of the first representative of each search-model ensemble derived from the largest cluster against the native structure. Successful structure solutions of individual search models are highlighted in blue and unsuccessful solutions in red.  Progressively darker shades of either colour correspond to increasing numbers of overlapping points.  Progressive truncation is shown for (a) ROSETTA and (b) PCONSC2+BBCONTACTS decoys (or PCONSC2 decoys for all-\textalpha\ targets).}
    \label{fig:ample_proof_mrglobtl}
\end{figure}

In comparison to the clustering step, the progressive truncation of decoys in the largest cluster at 20 different intervals directly affects the number of successful search models. An analysis of the progressive truncation and the effects on search model accuracy revealed that all successful search model ensembles had a C\textalpha-\gls{rmsd} better than 5.5\AA\ compared to the native structure (\cref{fig:ample_proof_mrglobtl}). Although the latter cutoff is independent of whether contact information was provided during \textit{ab initio} modelling, a clear difference between the ROSETTA and PCONSC2+BBCONTACTS (or PCONSC2 for all-\textalpha) ensemble search models for all targets can be observed. In total, ROSETTA decoys for all targets produced 1,314 ensemble search models based on the largest clusters. In comparison, PCONSC2+BBCONTACTS decoys generated for the same targets 2,469 search model ensembles from the largest clusters. This increase is the result of a more successful subclustering process due to the increased structural homogeneity across the decoys in the largest cluster. The most notable difference between the two sets is detected for the Small G-protein ARF6-GDP (\gls{pdb} ID: 1e0s), which produced three ensemble search models based on ROSETTA decoys and 90 based on PCONSC2+BBCONTACTS decoys. Additionally, ensemble search models with structural fragments of 15-40 residues of the target sequence are more likely to succeed in \gls{mr} phasing than larger or smaller search models \cite{Bibby2012-lm}. Here we find that the same range is most successful for contact-assisted decoys (\cref{fig:ample_proof_mrglobcompl}). Out of 246 successful search models for PCONSC2+BBCONTACTS decoys derived from the largest cluster (PCONSC2 for all-\textalpha), 101 successful search models contained 15-40 residues. Significantly, some cases like the PH domain of TAPP1 (\gls{pdb} ID: 1eaz) and the N-terminal bromodomain of human BRD4 (\gls{pdb} ID: 4cl9) only solved with truncated search models in this size range. Nevertheless, structure solutions were also achieved with larger or smaller search models. The smallest search model leading to a structure solution contained nine residues (8\% of total sequence) and solved the Calponin Homology domain from human \textbeta-spectrin (\gls{pdb} ID: 1bkr). In comparison, the largest successful search model in terms of residues was found for the designed full consensus ankyrin (\gls{pdb} ID: 2qyj) domain with 158 residues (95\% of total), and in terms of percentage of the total sequence the untruncated, 62 residue search model for \textalpha-spectrin SH3 domain (\gls{pdb} ID: 2nuz) was successful. Therefore, although truncating the \textit{ab initio} decoys at different levels remains essential for contact-assisted decoys, biasing sampling into the most successful size range may be advantageous in future runs.

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{ample_proof_mrglobcompl.pdf}
    \caption[Summary of AMPLE truncation ranges for structure solution]{Summary of AMPLE truncation ranges for structure solution. (a) Percentage of residues and (b) number of residues per chain in search model mapped against the number of search models leading to structure solution (blue) or not (red).}
    \label{fig:ample_proof_mrglobcompl}
\end{figure}

The truncated decoys are further processed by subclustering at three different atomic radii, with the resulting subclusters previously found to be similarly successful \cite{Bibby2012-lm}. Similar trends are seen here: 36\% of structure solutions with ROSETTA decoys were achieved with a subclustering radius of 1\AA, 36\% at a radius of 2\AA, and 28\% at a radius of 3\AA. For PCONSC2+BBCONTACTS (or PCONSC2 for all-\textalpha) decoy sets similar numbers were observed (35\% at radius of 1\AA; 40\% at 2\AA; 25\% at 3\AA). Nevertheless, in terms of number of targets solved all three subclustering radii were essential. Largest-cluster decoys for target 1eaz produced a total of 327 search models, but only one solved and this derived from a subclustering radius of 1\AA. In comparison, contact-assisted decoys from the largest cluster for target 4u3h achieved structure solutions solely with decoys subclustered at 2\AA. A single search model with subclustering radius of 3\AA\ solved the target 4cl9 with ROSETTA decoys. The final step in search model creation is the side-chain processing of each subclustered ensemble. Similarly to the subclustering, no difference was observed between ROSETTA and PCONSC2+BBCONTACTS decoys. For both the polyalanine treatment is most successful, covering 37\% of successful search models for ROSETTA decoys and 44\% for PCONSC2+BBCONTACTS decoys. For almost all targets, the polyalanine side-chain treatment would be enough to obtain a structure solution.  However, some cases, like the target 1eaz, only solve with either or both of the remaining treatments.  Thus, relying solely on polyalanine side-chain treatment may limit the overall success rate, although trialling polyalanine ensemble search models first might lead to structure solution faster.

\subsubsection{Transmembrane protein targets}
The \gls{mr} structure solution attempts given the decoy sets for transmembrane protein targets was conducted by Dr Jens Thomas and is documented in \textcite{Thomas2017-qu} and \textcite{Thomas2017-sh}.

In summary, \gls{mr} structure solution successes with decoys restrained by either of the three contact prediction protocols --- CCMPRED, MEMBRAIN and METAPSICOV STAGE1 --- were mixed. CCMPRED solved three targets, MEMBRAIN solved
five and METAPSICOV STAGE1 decoys solved four. Simple ROSETTA decoys resulted in four structure solutions. CCMPRED and METAPSICOV STAGE1 both solved target 4dve, which could not be solved with any other method, and METAPSICOV STAGE1 also solved target 2o9g, which had previously only been solved with the AMPLE library of ideal helices.

\section{Discussion}
The change in statistical model for residue-residue contact prediction has enabled great improvements to its precision. Today, contact information is often used to restrain the conformational search space to enable accurate \textit{ab initio} protein structure prediction. In this study, the effect of such improved structure predictions was examined with a particular interest of their application in unconventional \gls{mr} in AMPLE. The main focus of the presented work rests with the aim to extend AMPLE's target tractability, both for larger and more \textbeta-rich protein targets.

The addition of predicted residue-residue contacts unsurprisingly improved the quality of \textit{ab initio} protein structure predictions, which is in line with numerous previous studies \cite[e.g.,][]{Marks2011-os,Michel2014-eg,Kosciolek2014-bt,Ovchinnikov2015-tn,Ovchinnikov2016-jj,Michel2017-xh,De_Oliveira2017-sg,Ovchinnikov2017-nd,Wang2017-rx}. The improved decoy quality directly translates to further structure solutions with AMPLE. Contact-unassisted decoys, i.e. the current default, achieved nine and four solutions for globular and transmembrane protein targets, respectively. In comparison, contact-assisted decoys solve a further five globular targets, whilst contact-assisted decoys solved some different targets compared to contact-unassisted decoys for transmembrane protein targets.

The initial findings in this study highlighted the successful application of contact prediction to extend the target tractability with regards to the target chain length. \textcite{Bibby2012-lm} previously benchmarked \textit{ab initio} protein structure predictions up to chain lengths of 120 residues. However, the findings indicated that larger targets should be tractable with AMPLE, especially all-\textalpha\ ones \cite{Bibby2012-lm}. In this study, we confirm such extended target tractability, with contact-unassisted decoys leading to structure solutions up to 213 residues for globular targets and 223 residues for transmembrane protein targets. The addition of contacts to limit the conformational search space enabled structure solutions for the largest target in the globular target dataset with a 221-residue chain length and the transmembrane dataset with a 249-residue chain length. The fact that both of these targets are the largest in their sets is highly suggestive that contact-assisted decoys may enable solutions for much larger targets. In fact, recent research highlighted the successful \textit{ab initio} structure prediction of globular and transmembrane protein targets with target chain lengths in excess of 300 residues \cite{Ovchinnikov2017-nd}, which further supports this claim.

AMPLE was previously also limited by the target fold \cite{Bibby2012-lm}. Whilst the majority of all-\textalpha\ protein targets were comfortably tractable, mixed \textalpha-\textbeta\ and all-\textbeta\ targets were not \cite{Bibby2012-lm}. This limitation primarily arose from upstream limitations in \textit{ab initio} protein structure prediction but also the challenging task of tracing \textbeta-sheets in SHELXE, which was used to assess the successful structure solution. The use of contact-assisted decoys in AMPLE improved the target tractability for \textbeta-structure-containing protein targets. Structure solutions for four additional, \textbeta-structure-containing targets were obtained when contact-assisted decoys were used. A novel approach of combining \textbeta-sheet-specific contact pairings with a normal base prediction enabled the structure solution of one further target. Although no \gls{mr} structure solutions were lost when BBCONTACTS contact pairs were added to a base set of contact restraints, further studies are required to support routine application in AMPLE. Furthermore, BBCONTACTS contact pairs were identified by analysis of a CCMPRED predicted contact map, which is generally much noisier than metapredictor alternatives. Thus, further studies may explore the benefits or drawbacks of BBCONTACTS based on alternative predicted contact maps. Lastly, since the release of BBCONTACTS, other \textbeta-strand specific contact identification protocols have been developed \cite{Mao2018-mo}, which may need to be explored too.

Beyond the proof-of-concept study outlined in this chapter, it is very important to appreciate new limitations and unexplored areas of this work. At the time of conducting this study, PCONSC2 proved to be the state-of-the-art metapredictor. However, numerous alternatives have since been developed with more advanced Machine Learning architectures to post-process multiple individual contact predictions \cite[e.g.,][]{Jones2015-vq,Ovchinnikov2015-tn}. Furthermore, the optimal introduction of contacts as distance restraints into \textit{ab initio} protein structure prediction protocols is not yet clearly defined, and thus leaves the choice to the user without much comparison or guidance as to which works best. Lastly, contact information was exclusively used to restrain the \textit{ab initio} protein structure prediction procedure despite other potential applications in the AMPLE cluster-and-truncate algorithm. Subsequent chapters therefore explore additional uses of contact information for obtaining more accurate structure predictions (\cref{chap:rosetta_energy_functions}), identifying the implications on different structure prediction protocols (\cref{chap:alternate_abinitio_protocols}), and establishing improved decoy selection for better AMPLE processing (\cref{chap:ample_decoys}).
